packages/audiofileplayer/lib/audio_system.dart

```diff
@@ -0,0 +1,380 @@
+/// used, and the [shouldPlayWhileAppPaused] flag must also be set.
+///
+/// See
+/// https://developer.apple.com/documentation/avfoundation/avaudiosessioncategory
+/// for more information.
+enum IosAudioCategory {
+  /// Audio is silenced by screen lock and the silent switch; audio will not mix
+  /// with other apps' audio.
+  ambientSolo,
+
+  /// Audio is silenced by screen lock and the silent switch; audio will mix
+  /// with other apps' (mixable) audio.
+  ambientMixed,
+
+  /// Audio is not silenced by screen lock or silent switch; audio will not mix
+  /// with other apps' audio.
+  ///
+  /// The default value.
+  playback
+}
+
+/// A button to be added to the Android notification display via
+/// [setAndroidMediaButtons].
+///
+/// These are the 'built-in' buttons, with icons provided by this plugin. Users
+/// may add additional custom buttons with [AndroidCustomMediaButton].
+///
+/// These will yield [MediaEvent]s with the equivalent [MediaActionType].
+enum AndroidMediaButtonType {
+  stop,
+  pause,
+  play,
+  next,
+  previous,
+  seekForward,
+  seekBackward,
+}
+
+/// A custom button to be added to the Android notification display via
+/// [setAndroidMediaButtons].
+class AndroidCustomMediaButton {
+  const AndroidCustomMediaButton(
+      this.title, this.eventId, this.drawableResource);
+
+  final String title;
+
+  /// A unique identifier for the button, which will be returned the
+  /// [MediaEvent.customEventId].
+  final String eventId;
+
+  /// The name of the Android icon; must be in the Android project's "drawable"
+  /// resource folder.
+  final String drawableResource;
+}
+
+/// Metadata, used for display in the OS background audio system.
+class AudioMetadata {
+  const AudioMetadata({
+    this.id,
+    this.title,
+    this.album,
+    this.artist,
+    this.genre,
+    this.durationSeconds,
+    this.artBytes,
+  });
+  final String id;
+  final String title;
+  final String artist;
+  final String album;
+  final String genre;
+  final double durationSeconds;
+  final Uint8List artBytes;
+}
+
+/// Sends information to the OS's background audio system.
+///
+/// Only call these methods if you have an [Audio] to play in the background,
+/// and you wish to inform the OS on properties for display/interaction, i.e.
+/// showing metadata, and supporting interaction from the lockscreen/
+/// control-center (iOS), notification (Android), or external controllers like
+/// bluetooth, watch, auto.
+class AudioSystem {
+  AudioSystem._internal();
+
+  static final AudioSystem _instance = AudioSystem._internal();
+  static AudioSystem get instance => _instance;
+
+  /// Note that this currently shares the same [MethodChannel] and
+  /// native implementations as [Audio], with method call handler in [Audio].
+  /// In the future this will be split into multiple [MethodChannel]s and native
+  /// implementation classes.
+  static final MethodChannel channel = const MethodChannel(channelName);
```

Comment:
Nit: You can move MethodChannel definition to top level so it can be shared between classes.

There's no 1-1 mapping between Dart classes, Method channels and Platform classes. MethodChannel is an implementation detail that the application do not need to know about (which is why I was against application test code creating mock method call handler). All we care is that a union of API sets A, B and C is supported via method channel M. Assigning separate method channels to A, B and C is valid but it does complicate registration and maintenance.

Reply:
Done, PTAL